src/backend/d3d12/CommandBufferD3D12.cpp

```diff
@@ -78,18 +80,73 @@ namespace d3d12 {
                 case Command::CopyBufferToBuffer:
                     {
                         CopyBufferToBufferCmd* copy = commands.NextCommand<CopyBufferToBufferCmd>();
+                        auto src = ToBackend(copy->source.buffer.Get())->GetD3D12Resource();
+                        auto dst = ToBackend(copy->destination.buffer.Get())->GetD3D12Resource();
+                        commandList->CopyBufferRegion(dst.Get(), copy->destination.offset, src.Get(), copy->source.offset, copy->size);
                     }
                     break;
 
                 case Command::CopyBufferToTexture:
                     {
                         CopyBufferToTextureCmd* copy = commands.NextCommand<CopyBufferToTextureCmd>();
+                        auto buffer = ToBackend(copy->source.buffer.Get());
+                        auto texture = ToBackend(copy->destination.texture.Get());
+                        auto src = buffer->GetD3D12Resource();
+                        auto dst = texture->GetD3D12Resource();
+
+                        D3D12_TEXTURE_COPY_LOCATION srcLocation;
+                        srcLocation.pResource = src.Get();
+                        srcLocation.Type = D3D12_TEXTURE_COPY_TYPE_PLACED_FOOTPRINT;
+
+                        D3D12_TEXTURE_COPY_LOCATION dstLocation;
+                        dstLocation.pResource = dst.Get();
+                        dstLocation.Type = D3D12_TEXTURE_COPY_TYPE_SUBRESOURCE_INDEX;
+                        dstLocation.SubresourceIndex = copy->destination.level;
+
+                        auto desc = dst->GetDesc();
+                        desc.Width = copy->destination.width;
+                        desc.Height = copy->destination.height;
+                        desc.DepthOrArraySize = copy->destination.depth;
+                        uint64_t totalBytes;
+                        device->GetD3D12Device()->GetCopyableFootprints(&desc, copy->destination.level, 1, copy->source.offset, &srcLocation.PlacedFootprint, nullptr, nullptr, &totalBytes);
```

Comment:
Ideally I'd like us to fill this data manually as we should have all the required information present in our state tracking. This would help understand exactly what's going on.